src/applications/vaos/reducers/newAppointment.js

```diff
@@ -457,11 +457,19 @@ export default function formReducer(state = initialState, action) {
       return {
         ...state,
         pastAppointments: action.appointments,
+        data: {
+          ...state.data,
+          calendarData: {},
+        },
```

Comment:
Clears out `calendarData` with `START_REQUEST_APPOINTMENT_FLOW` or `START_DIRECT_SCHEDULE_FLOW`.  Previously, when a user selected a date, went back and switched flows, the `calendarData` remained populated and they were able to bypass validation.